src/mainwindow.cpp

```diff
@@ -740,3 +738,16 @@ void MainWindow::addNewTab()
     else
         consoleTabulator->addNewTab();
 }
+
+void MainWindow::onCurrentTitleChanged(int index)
+{
+    QString title;
+    QIcon icon;
+    if (-1 != index)
+    {
+        title = consoleTabulator->tabText(index);
+        icon = consoleTabulator->tabIcon(index);
+    }
+    setWindowTitle(title.isEmpty() ? QStringLiteral("QTerminal") : title);
+    setWindowIcon(icon.isNull() ? QIcon::fromTheme("utilities-terminal") : icon);
```

Comment:
Maybe this line should be removed. For example if I run ```mpv xxx.mp4``` from QTerminal, the window icon becomes the same as mpv's. It brings inconvenience in LXQT, as I use icons in the Alt-Tab window to tell different applications.